why-django.qmd

```diff
@@ -54,99 +54,80 @@ and drawbacks for each option.
+- Promotes security best-practices through the built-in authentication system, role-based access control, fine-grained permissions, Cross-Site Request Forgery (CSRF) protection and password hashing.
+- Provides a powerful Object-Relational Mapper (ORM) that abstracts away much of the database logic and allows developers to work with database models in a Pythonic way.
+- Supports all major databases and has a migration mechanism, making it easy to evolve the database schema over time.
+- Has good file storage support, making it easier to work with raw data and image files.
+- Comes with an admin interface out of the box, allowing non-technical users to manage data.
+- Offers built-in form handling capabilities with customisable form classes and form validation logic.
+- Uses server-side rendering by default, enabling fast webpage renders. 
+- Has a template engine for creating dynamic HTML pages.
+- Makes it easy to compose Django applications and has a wide array of well-maintained packages.
+- Has excellent documentation and a large and active developer community.
 :::
 ::: {.column}
 #### Drawbacks
 
-- Item 1
+- Has a steeper learning curve because of its extensive set of features.
+- May be less performant in certain scenarios than the other frameworks considered due to the size of the core framework and its limited support for asynchronous programming.
+- More opinionated than other frameworks, with less flexibility about the choice of project structure, ORM and template engine.
+- Server-rendered websites can have slower response times, which can limit interactivity.
+- Using the built-in template engine creates a tighter coupling between backend and frontend, potentially complicating code reuse and maintenance.
 :::
 :::
 
 ### Flask
 
-[Flask](https://flask.palletsprojects.com/en/2.3.x/) is a lightweight
-web framework that is easy to use and has a low learning curve. It
-provides a lot of flexibility and does not enforce any particular
-project structure. Flask has a small footprint and is a good option for
-simple web applications or prototypes.
-
-However, Flask does not provide some of the advanced features that
-Django has, such as an admin interface, an object-relational mapper
-(ORM), and a built-in authentication system. These features would need
-to be implemented manually or by using third-party libraries.
-
+[Flask](https://flask.palletsprojects.com/en/3.0.x/) is a lightweight web framework, designed to be simple and flexible.
 
 ::: {.columns}
 ::: {.column}
 #### Benefits
 
-- Item 1
+- Has a gentler learning curve and is easy to use.
+- Provides a lot of flexibility without enforcing a particular project structure, ORM or frontend solution.
+- Has a smaller footprint, which may lead to a small performance gain.
+- Uses Jinja2 as its default template engine.
```

Comment:
This might just be my ignorance, but why's using Jinja2 a benefit? :)

Reply:
Ah sorry, I meant that the benefit is that is has a default one at all

Reply:
Ahhh! Maybe write that a bit more explicit then :)